packages/discovery/src/discovery/config/ConfigReader.ts

```diff
@@ -3,19 +3,26 @@ import path, { posix } from 'path'
 import { assert } from '@l2beat/backend-tools'
 import { DiscoveryOutput } from '@l2beat/discovery-types'
 import { readFile } from 'fs/promises'
-import { ParseError, parse } from 'jsonc-parser'
 
 import { stripAnsiEscapeCodes } from '@l2beat/shared-pure'
 import chalk from 'chalk'
 import { ZodError } from 'zod'
 import { fileExistsCaseSensitive } from '../../utils/fsLayer'
+import { TemplateService } from '../analysis/TemplateService'
+import { readJsonc } from '../utils/readJsonc'
 import { DiscoveryConfig } from './DiscoveryConfig'
 import { DiscoveryMeta } from './DiscoveryMeta'
-import { DiscoveryContract, RawDiscoveryConfig } from './RawDiscoveryConfig'
+import { RawDiscoveryConfig } from './RawDiscoveryConfig'
 
 export const TEMPLATES_PATH = path.join('discovery', '_templates')
 
 export class ConfigReader {
+  public templateService: TemplateService
+
+  constructor() {
+    this.templateService = new TemplateService()
+  }
+
```

Comment:
Do we really need an entire object? I'm starting to think that `TemplateService` should really be just a couple of functions. The connection between `ConfigReader` and `TemplateService` is really hard to understand.

- `ConfigReader` still exports `TEMPLATES_PATH`, that should probably be in the `TemplateService`
- But `TEMPLATES_PATH` is only used to read the correct files, reading now is split between two classes - `ConfigReader` and `TemplateService` - some files are read by one and others are read by the other

I think that this is just a big mess. I'm currently trying to refactor the project to use `ConfigReader` exclusively and it's really hard because it's not entirely contained in itself. The fact that it's split like that makes this code _very_ rigid, it just **has** to be used from a given directory and any attempt to make it support reading from a different place is a nightmare.

It feels like `TemplateService` has no well defined scope of things it's responsible for without taking some responsibility from `ConfigReader`. My suggestion is to make `ConfigReader` a base class (not in the inheritance kind of way) that does only reading. Any class/function that wants to read some files from the `discovery/project` subdirectories **has** to go through `ConfigReader`.